diagnostic.md

```diff
@@ -5,74 +5,97 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```md
-// your response here
+The backend is used to store data, send data, and process requests sent from a client or front-end server. It is used to help do the processing of applications and allow for "persistent state" storage.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```md
-// your response here
+The controller uses the Model layer to fetch data
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```md
-// your response here
+The Controller communicates with the Model.
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```md
-// your response here
+We don't use view since our browser or client acts as the view layer.  It interprets the data
+sent and renders it in the browser.
 ```
 
 What does C.R.U.D stand for?
 
 ```md
-// your response here
+Create , Read, Update, Destroy/Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```md
-// your response here
+We see them in the Controller view.
 ```
 
 List at least 5 standard rails actions that C.R.U.D requests correspond to?
 
 ```md
-// your response here
+Create - create
+Read - index , show
+Update - update
+Destroy/ Delete - destroy
 ```
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```md
-// your response here
+1) The Client sends out a GET request to the server for the item at '/people/1'
+2) The server acts as a router and goes to the People controller to see if they can find this request for the item at '/people/1'
+3) The People Controller then approaches the model to retrieve the item at '/people/1'
+4) The People Model looks for the item and either lets the People controller know that it found the item or it did not
+5) The People Controller then takes this information (found or not found) and passes on to the server
+6) Finally, the server either returns to the user the result of '/people/1' or a notification letting them know that they could not find the item.
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+bin/rails new my_api
 ```
 
 What is the command to start an instance of a rails server?
 
 ```bash
-// your response here
+bin/rails server
 ```
 
 What are the commands to drop, create, migrate and seed a database from the command
 line? (5 bullet points)
 
 ```bash
-// your response here
+* bin/rake db:drop
+* bin/rake db:create
+* bin/rake db:migrate
+* bin/rake db:seed
+* bin/rake db:examples -- to populate examples.
 ```
 
 What is the command to scaffold a pet with a name and age attributes (hint:
 Also think of the data types for each attribute)?
 
 ```bash
-// your response here
+class Pets < ActiveRecord::Migration[5.0]
+  def change
+    create_table :pets do |t|
+      t.string :name
+      t.integer :age
+
+      t.timestamps
+    end
+  end
+end
+
```

Comment:
I see you got a bit confused by this one. We were looking for the command you would enter into the terminal to scaffold the pet entity for us which would look like:

`rails generate scaffold pet name:string age:integer`